component/http/middleware.go

```diff
@@ -131,6 +140,147 @@ func NewLoggingTracingMiddleware(path string) MiddlewareFunc {
 	}
 }
 
+type compressionResponseWriter struct {
```

Comment:
Question: Since compression is chosen with a header maybe we do not need to let the user chose the method. We support all of them and if the header contains the correct value we compress using the requested method.
This could also mean that the middle-ware could be enabled by default and no need to opt in.

Reply:
not sure if it can be enabled by default, i think - and i might be mistaken - it depends also on the client.

Reply:
That's a great idea, it totally flew over my head. I've prepared a commit that does just that; it populates the builder with some default parameters for Deflate and LZW, and utilizes `SetDeflateParams` and `SetLZWParams` methods if the user wants to override them.

After some more polish and test fixing I'll push it so we can see it in action. I don't believe that it would add much weight to people who want to just skip the compression, so we could think about adding it to the default HTTP component.